### PR TITLE
Update Embedding Generation Code

### PR DESCRIPTION
**Pull request by amogkam**
Addresses feedback from https://github.com/ray-project/langchain-ray/pull/2

### CODE REVIEW

File: embedding_pdf_documents/requirements.txt [hunk 1]
```diff
@@ -0,0 +1,7 @@
+langchain
+pypdf
+ray[data]
+torch
+sentence-transformers
+transformers
+faiss-cpu
```
Comment (kamil-kaczmarek) [inline embedding_pdf_documents/requirements.txt]:
```suggestion
arxiv
faiss-cpu
langchain
pypdf
ray[data]
sentence-transformers
torch
transformers
```

Comment (kamil-kaczmarek) [reply embedding_pdf_documents/requirements.txt]: added arxiv

File: embedding_pdf_documents/embedding_ray.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        try:
+            text.append(page.extract_text())
+        except binascii.Error:
+            # Skip all pages that are not parseable due to malformed characters.
+            print("parsing failed")
+    return text
+    
+# We use `flat_map` as `convert_to_text` has a 1->N relationship.
+# It produces N strings for each PDF (one string per page).
+# Use `map` for 1->1 relationship.
```
Comment (waleedkadous) [inline embedding_pdf_documents/embedding_ray.py:38]: Beautiful Explanation!

**Review by waleedkadous — COMMENTED** [review]
The new comments are super helpful. Thank you for taking the time to add them!

File: embedding_pdf_documents/README.md [hunk 3]
```diff
@@ -1,9 +1,13 @@
-# Creating a Ray cluster with 20 GPUs
+# Scaing Embedding Generation with LangChain and Ray
```
Comment (kamil-kaczmarek) [inline embedding_pdf_documents/README.md]:
```suggestion
# Scaling Embedding Generation with LangChain and Ray
```

File: embedding_pdf_documents/embedding_ray.py [hunk 4]
```diff
@@ -85,12 +102,12 @@ def __call__(self, text_batch: List[str]):
 for output in ds.iter_rows():
     text_and_embeddings.append(output)
 
-vectore_store = FAISS.from_embeddings(
+vectore_stor = FAISS.from_embeddings(
```
Comment (kamil-kaczmarek) [inline embedding_pdf_documents/embedding_ray.py]:
```suggestion
vector_store = FAISS.from_embeddings(
```

File: embedding_pdf_documents/embedding_ray.py [hunk 5]
```diff
@@ -85,12 +102,12 @@ def __call__(self, text_batch: List[str]):
...
     text_and_embeddings,
     # Provide the embedding model to embed the query.
     # The documents are already embedded.
     embedding=HuggingFaceEmbeddings(model_name=model_name),
 )
 
 # Persist the vector store.
-vectore_store.save_local("faiss_index")
+vectore_stor.save_local("faiss_index")
```
Comment (kamil-kaczmarek) [inline embedding_pdf_documents/embedding_ray.py]:
```suggestion
vector_store.save_local("faiss_index")
```

**Review by kamil-kaczmarek — CHANGES_REQUESTED** [review]
typos

**Review by kamil-kaczmarek — APPROVED** [review]

**Review by waleedkadous — APPROVED** [review]

**Pull request merged**